src/components/AssignmentCreatorForm.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {t} from 'i18next';
+import classnames from 'classnames';
+import map from 'lodash-es/map';
+import chrono from 'chrono-node';
+
+import PropTypes from 'prop-types';
+import {Field, reduxForm} from 'redux-form/immutable';
+
+import {dateToString} from '../util/timeFormatter';
+
+import renderField from './renderField';
+import renderSelectField from './renderSelectField';
+
+function validate(values) {
+  const errors = {};
+  const input = values.get('date');
+  const parsedResponse = chrono.parse(input);
+
+  if (!values.get('course')) {
+    errors.course = '* Course Required';
+  } else if (values.get('course') === '') {
+    errors.course = 'Must Select a Course';
+  }
+  if (!values.get('date')) {
+    errors.date = '* Date Required';
+  }
+  if (parsedResponse[0]) {
+    const parsedDate = parsedResponse[0].start.date();
+    if (parsedDate < Date.now()) {
+      errors.date = 'Date must be in the future :)';
+    }
+  }
+  return errors;
+}
+
+function warn(values) {
+  const warnings = {};
+  const input = values.get('date');
+  const parsedResponse = chrono.parse(input);
+  if (parsedResponse[0]) {
+    const parsedDate = parsedResponse[0].start.date();
+    const dateString = dateToString(parsedDate);
+    warnings.date = dateString;
+  }
+  return warnings;
+}
+
+function AssignmentCreatorForm({
+  courses,
+  handleSubmit,
+  onAssignAssignment,
+  onDraftAssignment,
+  onCloseAssignmentCreator,
+}) {
+  const courseOptions = map(courses, course => (
+    <option key={course.id} value={course.id}>{course.name}</option>
+  ));
```

Comment:
These days I am more into inlining everything into the JSX and avoiding intermediate variables, but I don’t feel strongly about it.